rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -217,6 +223,22 @@ void Player::prepare_clock(const PlayOptions & options, rcutils_time_point_value
 {
   double rate = options.rate > 0.0 ? options.rate : 1.0;
   clock_ = std::make_unique<rosbag2_cpp::TimeControllerClock>(starting_time, rate);
```

Comment:
I know not related to the change, but why is the clock heap allocated?

Reply:
I was thinking about deferred initialization - we don't have enough information to create the clock in the constructor, as it is right now. I could just overwrite a stack-allocated member here, alternatively. Thoughts?

Reply:
I might address this in #716